src/main/java/org/wildfly/security/sasl/otp/OTPPasswordAndParameterCallbackHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.security.sasl.otp;
+
+import java.io.IOException;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+
+import org.wildfly.security.auth.callback.ParameterCallback;
+import org.wildfly.security.password.spec.OneTimePasswordAlgorithmSpec;
+
+/**
+ * A callback handler for OTP which handles {@link ParameterCallback} and {@link PasswordCallback}
+ *
+ * @author Kabir Khan
+ */
+public class OTPPasswordAndParameterCallbackHandler implements CallbackHandler {
+    private final PasswordFormat passwordFormat;
+    private final String password; // The pass phrase or OTP
+    private final String newAlgorithm;
+    private final byte[] newSeed;
+    private final int newSequenceNumber;
+    private final PasswordFormat newPasswordFormat;
+    private final String newPassword; // The new pass phrase or new OTP
```

Comment:
I don't think we should allow this.  We could, for example, allow a callback handler to reset normal passwords too, however that seems equally inappropriate to me.